File: mobile_client/lib/voice/orb_ticker_policy.dart
/// The orb runs a continuous 60fps blurred/additive CustomPainter. Because every
/// page lives forever in NavShell's IndexedStack, its ticker would otherwise keep
/// repainting on EVERY tab. It only needs to animate while its OWNING tab is the
/// active tab (the orb appears on both the Voice tab and the Chat empty-state).
/// Flutter already mutes tickers when the app is backgrounded. A null [ownerTab]
/// means "always animate" (not nav-gated). Pure so it's unit-testable.
bool orbTickerEnabled({required int activeTab, required int? ownerTab}) =>
    ownerTab == null || activeTab == ownerTab;

File: ios_app/JarvisCopilot/Copilot/Voice/OrbTickerPolicy.swift
import Foundation

/// The orb runs a continuous 60 fps blurred/additive draw. Because every page
/// lives forever in the tab shell, its ticker would otherwise keep repainting on
/// EVERY tab. It only needs to animate while its OWNING tab is the active tab
/// (the orb appears on both the Voice tab and the Chat empty-state). SwiftUI
/// already mutes `TimelineView` when the app is backgrounded. A nil `ownerTab`
/// means "always animate" (not nav-gated). Pure so it's unit-testable.
/// Port of `voice/orb_ticker_policy.dart`.
func orbTickerEnabled(activeTab: Int, ownerTab: Int?) -> Bool {
    ownerTab == nil || activeTab == ownerTab
}
